### PR TITLE
Prokofev Kirill Lab1 Var4

### PR DESCRIPTION
**Pull request by Kirill23323**
The plugin changes the identifiers of functions, variables, and classes.

### CODE REVIEW

File: clang/labs/lab1/prokofev_kirill/RenameIdentificator.cpp [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    return std::make_unique<RenameASTConsumer>(CI, type, oldName, newName);
+  }
+
+protected:
+  bool ParseArgs(const clang::CompilerInstance &CI,
+                 const std::vector<std::string> &args) override {
+    std::vector<std::pair<std::string, std::string>> params = {
+        {"type=", ""}, {"oldName=", ""}, {"newName=", ""}};
+
+    if (!args.empty() && args[0] == "help") {
```
Comment (m-ly4) [inline clang/labs/lab1/prokofev_kirill/RenameIdentificator.cpp]: What if `help` arg is not first?

File: clang/labs/lab1/prokofev_kirill/RenameIdentificator.cpp [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+      PrintHelp(llvm::errs());
+      return false;
+    }
+
+    if (args.size() < 3) {
```
Comment (m-ly4) [inline clang/labs/lab1/prokofev_kirill/RenameIdentificator.cpp]: What if number of args > 3?

**Review by m-ly4 — CHANGES_REQUESTED** [review]

File: clang/labs/lab1/prokofev_kirill/RenameIdentificator.cpp (hunk 1 above)
Comment (Kirill23323) [reply clang/labs/lab1/prokofev_kirill/RenameIdentificator.cpp]: An argument error will be displayed.

File: clang/labs/lab1/prokofev_kirill/RenameIdentificator.cpp (hunk 2 above)
Comment (Kirill23323) [reply clang/labs/lab1/prokofev_kirill/RenameIdentificator.cpp]: An argument error will be displayed.

File: clang/test/lab1/prokofev_kirill/test.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+// CLASS-NEXT:  Beta(){};
+// CLASS-NEXT:  ~Beta();
+// CLASS-NEXT:};
+
+//--- class_test.cpp
+class Alpha {
+public:
+  Alpha(){};
+  ~Alpha();
+};
```
Comment (aobolensk) [inline clang/test/lab1/prokofev_kirill/test.cpp]: Please, extend test suite with more cases: function definitions outside the class, variables that have "Alpha"  as a substring in their names and so on

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: clang/test/lab1/prokofev_kirill/test.cpp (hunk 3 above)
Comment (Kirill23323) [reply clang/test/lab1/prokofev_kirill/test.cpp]: Updated the class test

File: clang/labs/lab1/prokofev_kirill/RenameIdentificator.cpp (hunk 2 above)
Comment (m-ly4) [reply clang/labs/lab1/prokofev_kirill/RenameIdentificator.cpp]: I specify required arguments and `help` argument in the end. So, total number of args is 4, help is present. I expect to see help message only.

Comment (Kirill23323) [reply clang/labs/lab1/prokofev_kirill/RenameIdentificator.cpp]: Fixed the check and added the appropriate test

**Review by aobolensk — APPROVED** [review]

**Pull request merged**